src/triage/component/architect/label_generators.py

```diff
@@ -8,17 +9,40 @@
 )
 
 
-class BinaryLabelGenerator(object):
-
+class BinaryLabelBase(object, metaclass=ABCMeta):
     def __init__(self, events_table, db_engine):
         self.events_table = events_table
         self.db_engine = db_engine
 
     def validate(self):
         table_should_have_data(self.events_table, self.db_engine)
-        column_should_be_intlike(self.events_table, 'entity_id', self.db_engine)
-        column_should_be_timelike(self.events_table, 'outcome_date', self.db_engine)
-        column_should_be_booleanlike(self.events_table, 'outcome', self.db_engine)
+        column_should_be_intlike(
```

Comment:
These validations would only apply to the binary one anyway, but I think putting validations here has been superceded by the experiment validation class (src/triage/experiments/validate.py)